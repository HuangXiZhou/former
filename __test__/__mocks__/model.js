export default {
  linkage_foo: '',
  linkage_bar: '',
  none: '',
  input: '',
  textarea: '',
  select: 'bar',
  inputNumber: '',
  radio: 'foo',
  checkbox: 'foo',
  switch: false,
  slider: 0,
  datePicker: '',
  timeSelect: '',
  timePicker: '',
  tree: '',
  rate: null,
  cascader: [],
  upload: '',
  custom: ''
};
