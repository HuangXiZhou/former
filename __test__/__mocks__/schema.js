export default {
  'element-ui': [
    {
      label: 'linkage_foo',
      name: 'linkage_foo',
      type: 'input',
      class: [ 'likage_foo' ],
      onChange: (e, model) => {
        model.linkage_bar = e;
      }
    },
    {
      label: 'linkage_bar',
      name: 'linkage_bar',
      type: 'input',
      class: [ 'likage_bar' ]
    },
    {
      label: 'none',
      name: 'none',
      type: 'input',
      when: false
    },
    {
      label: 'input',
      name: 'input',
      type: 'input',
      top: 'top',
      prefix: 'prefix',
      suffix: 'suffix',
      bottom: 'bottom',
      htmlAttr: { placeholder: 'input placeholder' },
      rules:
      [
        { required: true, trigger: 'blur' }
      ],
      when: true
    },
    {
      label: 'textarea',
      name: 'textarea',
      type: 'textarea',
      top: (h) => h('div', 'top'),
      prefix: (h) => h('div', 'prefix'),
      suffix: (h) => h('div', 'suffix'),
      bottom: (h) => h('div', 'bottom'),
      htmlAttr: { placeholder: 'textarea placeholder' },
      when: () => true
    },
    {
      label: 'select',
      name: 'select',
      type: 'select',
      style: { width: '350px' },
      class: [ 'custom-class' ],
      options: [ {
        value: 'foo',
        label: 'foo'
      }, {
        value: 'bar',
        label: 'bar'
      } ]
    },
    {
      label: 'inputNumber',
      name: 'inputNumber',
      type: 'inputNumber',
    },
    {
      label: 'radio',
      name: 'radio',
      type: 'radio',
      options: [
        { value: 'foo', name: 'foo' },
        { value: 'bar', name:'bar' }
      ]
    },
    {
      label: 'checkbox',
      name: 'checkbox',
      type: 'checkbox',
      options: [
        { value: 'foo', name: 'foo' }
      ]
    },
    {
      label: 'switch',
      name: 'switch',
      type: 'switch'
    },
    {
      label: 'slider',
      name: 'slider',
      type: 'slider'
    },
    {
      label: 'datePicker',
      name: 'datePicker',
      type: 'datePicker'
    },
    {
      label: 'timeSelect',
      name: 'timeSelect',
      type: 'timeSelect'
    },
    {
      label: 'timePicker',
      name: 'timePicker',
      type: 'timePicker'
    },
    {
      label: 'tree',
      name: 'tree',
      type: 'tree',
      data: [ {
        label: 'foo',
        children: [ {
          label: 'bar',
          children: [ { label: 'foo' } ]
        } ]
      } ],
      defaultProps: {
        children: 'children',
        label: 'label'
      }
    },
    {
      label: 'rate',
      name: 'rate',
      type: 'rate'
    },
    {
      label: 'cascader',
      name: 'cascader',
      type: 'cascader',
      options: [ {
        value: 'foo',
        label: 'foo',
        children: [ {
          value: 'bar',
          label: 'bar'
        } ]
      } ]
    },
    {
      label: 'upload',
      name: 'upload',
      type: 'upload',
      action: 'https://foo.bar',
      slots: [ (h) => h('p', 'foo') ]
    },
    {
      label: 'custom',
      name: 'custom',
      type: 'custom',
      render: (h) => h('div', 'foo')
    }
  ],
  'iview': [
    {
      label: 'linkage_foo',
      name: 'linkage_foo',
      type: 'input',
      class: [ 'likage_foo' ],
      onChange: (e, model) => {
        model.linkage_bar = e;
      }
    },
    {
      label: 'linkage_bar',
      name: 'linkage_bar',
      type: 'input',
      class: [ 'likage_bar' ]
    },
    {
      label: 'none',
      name: 'none',
      type: 'input',
      when: false
    },
    {
      label: 'input',
      name: 'input',
      type: 'input',
      top: 'top',
      prefix: 'prefix',
      suffix: 'suffix',
      bottom: 'bottom',
      htmlAttr: { placeholder: 'input placeholder' },
      rules:
      [
        { required: true, trigger: 'blur' }
      ],
      when: true
    },
    {
      label: 'textarea',
      name: 'textarea',
      type: 'textarea',
      top: (h) => h('div', 'top'),
      prefix: (h) => h('div', 'prefix'),
      suffix: (h) => h('div', 'suffix'),
      bottom: (h) => h('div', 'bottom'),
      htmlAttr: { placeholder: 'textarea placeholder' },
      when: () => true
    }, {
      label: 'select',
      name: 'select',
      type: 'select',
      style: { width: '350px' },
      class: [ 'custom-class' ],
      options: [ {
        value: 'foo',
        label: 'foo'
      }, {
        value: 'bar',
        label: 'bar'
      } ]
    }, {
      label: 'inputNumber',
      name: 'inputNumber',
      type: 'inputNumber',
    },   {
      label: 'radio',
      name: 'radio',
      type: 'radio',
      options: [
        { value: 'foo', name: 'foo' },
        { value: 'bar', name:'bar' }
      ]
    }, {
      label: 'checkbox',
      name: 'checkbox',
      type: 'checkbox',
      options: [
        { value: 'foo', name: 'foo' }
      ]
    },
    {
      label: 'switch',
      name: 'switch',
      type: 'switch'
    },
    {
      label: 'slider',
      name: 'slider',
      type: 'slider'
    },
    {
      label: 'datePicker',
      name: 'datePicker',
      type: 'datePicker'
    },
    {
      label: 'timePicker',
      name: 'timePicker',
      type: 'timePicker'
    },
    {
      label: 'tree',
      name: 'tree',
      type: 'tree',
      data: [
        {
          title: 'foo',
          children: [
            {
              title: 'bar',
              children: [
                { title: 'foo' }
              ]
            }
          ]
        }
      ]
    },
    {
      label: 'rate',
      name: 'rate',
      type: 'rate'
    },
    {
      label: 'cascader',
      name: 'cascader',
      type: 'cascader',
      data: [ {
        value: 'foo',
        label: 'foo',
        children: [ {
          value: 'bar',
          label: 'bar'
        } ]
      } ]
    },
    {
      label: 'upload',
      name: 'upload',
      type: 'upload',
      action: 'https://foo.bar',
      slots: [ (h) => h('p', 'foo') ]
    }
  ]
};
