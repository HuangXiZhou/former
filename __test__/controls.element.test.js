import { createLocalVue, mount, config } from '@vue/test-utils';
import ElementUI from 'element-ui';
import Former from '../src/index';
import formerConfig from '../src/config';
import schema from './__mocks__/schema';
import model from './__mocks__/model';

config.stubs.transition = false;
const localVue = createLocalVue();
localVue.use(ElementUI);

describe('Former/form', () => {
  let formerInstance;

  beforeEach(() => {
    formerInstance = mount({
      data: () => ({
        ui: 'element-ui',
        options: {},
        schema: schema['element-ui'],
        model: model['element-ui']
      }),

      components: { Former },

      render(h) {
        return h('Former', {
          ref: formerConfig.namespace,
          props: {
            ui: this.ui,
            model: this.model,
            options: this.options,
            schema: this.schema
          }
        });
      }
    }, { localVue, sync: false });
  });

  it('Render `Input` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'input');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Textarea` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'textarea');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Select` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'select');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `InputNumber` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'inputNumber');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Radio` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'radio');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Checkbox` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'checkbox');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Switch` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'switch');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  // Will generate uniq value in aria-describedby
  // it('Render `Slider` component', () => {
  //   const _schema = schema['element-ui'].filter((v) => v.name === 'slider');
  //   formerInstance.setData({ schema: _schema });
  //   const html = formerInstance.html();
  //   expect(html).toMatchSnapshot();
  // });

  it('Render `DatePicker` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'datePicker');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `TimeSelect` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'timeSelect');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `TimePicker` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'timePicker');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Tree` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'tree');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Rate` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'rate');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Cascader` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'cascader');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Upload` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'upload');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });

  it('Render `Custom` component', () => {
    const _schema = schema['element-ui'].filter((v) => v.name === 'custom');
    formerInstance.setData({ schema: _schema });
    localVue.nextTick(() => {
      const html = formerInstance.html();
      expect(html).toMatchSnapshot();
    });
  });
});
