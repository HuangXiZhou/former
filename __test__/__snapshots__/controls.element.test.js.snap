// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Former/form Render \`Cascader\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="cascader" class="el-form-item__label">cascader</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main"><span class="el-cascader control"><div class="el-input el-input--suffix"><!----><input type="text" readonly="readonly" autocomplete="off" placeholder="请选择" class="el-input__inner"><!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-input__icon el-icon-arrow-down"></i><!----></span>
          <!----></span>
          <!---->
        </div><span class="el-cascader__label"></span></span>
      </div>
    </div>
    <!---->
  </div>
  </div>
</form>
`;

exports[`Former/form Render \`Checkbox\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="checkbox" class="el-form-item__label">checkbox</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div role="group" aria-label="checkbox-group" class="el-checkbox-group control"><label role="checkbox" class="el-checkbox"><span aria-checked="mixed" class="el-checkbox__input"><span class="el-checkbox__inner"></span><input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="foo"></span><span class="el-checkbox__label">foo<!----></span></label></div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`Custom\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="custom" class="el-form-item__label">custom</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div class="control">
            <div>foo</div>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`DatePicker\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="datePicker" class="el-form-item__label">datePicker</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--date control">
            <!----><input type="text" autocomplete="off" name="datePicker" class="el-input__inner"><span class="el-input__prefix"><i class="el-input__icon el-icon-date"></i><!----></span><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-input__icon"></i><!----></span>
            <!----></span>
            <!---->
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`Input\` component 1`] = `
<form class="el-form">
  <div class="el-form-item is-required"><label for="input" class="el-form-item__label">input</label>
    <div class="el-form-item__content">
      <div class="former-top">top</div>
      <div class="former-wrapper">
        <div class="former-prefix">prefix</div>
        <div class="former-main">
          <div class="el-input control">
            <!----><input type="text" autocomplete="off" aria-label="input" placeholder="input placeholder" class="el-input__inner">
            <!---->
            <!---->
            <!---->
          </div>
        </div>
        <div class="former-suffix">suffix</div>
      </div>
      <div class="former-bottom">bottom</div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`InputNumber\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="inputNumber" class="el-form-item__label">inputNumber</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div class="el-input-number control"><span role="button" class="el-input-number__decrease"><i class="el-icon-minus"></i></span><span role="button" class="el-input-number__increase"><i class="el-icon-plus"></i></span>
            <div class="el-input">
              <!----><input type="text" autocomplete="off" aria-label="inputNumber" max="Infinity" min="-Infinity" name="inputNumber" class="el-input__inner" role="spinbutton" aria-valuemax="Infinity" aria-valuemin="-Infinity" aria-valuenow="0" aria-disabled="false">
              <!---->
              <!---->
              <!---->
            </div>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`Radio\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="radio" class="el-form-item__label">radio</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div role="radiogroup" class="el-radio-group control"><label role="radio" aria-checked="true" tabindex="0" class="el-radio is-checked"><span class="el-radio__input is-checked"><span class="el-radio__inner"></span><input type="radio" aria-hidden="true" tabindex="-1" class="el-radio__original" value="foo"></span><span class="el-radio__label">foo<!----></span></label><label role="radio" tabindex="-1" class="el-radio"><span class="el-radio__input"><span class="el-radio__inner"></span><input type="radio" aria-hidden="true" tabindex="-1" class="el-radio__original" value="bar"></span><span class="el-radio__label">bar<!----></span></label></div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`Rate\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="rate" class="el-form-item__label">rate</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div role="slider" aria-valuenow="0" aria-valuetext="" aria-valuemin="0" aria-valuemax="5" tabindex="0" class="el-rate control" style="line-height: 2.4;"><span class="el-rate__item" style="cursor: pointer;"><i class="el-rate__icon el-icon-star-off" style="color: rgb(198, 209, 222);"><!----></i></span><span class="el-rate__item" style="cursor: pointer;"><i class="el-rate__icon el-icon-star-off" style="color: rgb(198, 209, 222);"><!----></i></span><span class="el-rate__item" style="cursor: pointer;"><i class="el-rate__icon el-icon-star-off" style="color: rgb(198, 209, 222);"><!----></i></span><span class="el-rate__item" style="cursor: pointer;"><i class="el-rate__icon el-icon-star-off" style="color: rgb(198, 209, 222);"><!----></i></span><span class="el-rate__item" style="cursor: pointer;"><i class="el-rate__icon el-icon-star-off" style="color: rgb(198, 209, 222);"><!----></i></span>
            <!---->
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`Select\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="select" class="el-form-item__label">select</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div class="el-select control custom-class" style="width: 350px;">
            <!---->
            <div class="el-input el-input--suffix">
              <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="请选择" name="select" class="el-input__inner">
              <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-arrow-up"></i><!----><!----></span>
              <!----></span>
              <!---->
            </div>
            <div class="el-select-dropdown el-popper" style="display: none;">
              <div class="el-scrollbar" style="">
                <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                  <ul class="el-scrollbar__view el-select-dropdown__list">
                    <!---->
                    <li class="el-select-dropdown__item">foo</li>
                    <li class="el-select-dropdown__item selected">bar</li>
                  </ul>
                </div>
                <div class="el-scrollbar__bar is-horizontal">
                  <div class="el-scrollbar__thumb" style="width: 0px; transform: translateX(0%); -webkit-transform: translateX(0%);"></div>
                </div>
                <div class="el-scrollbar__bar is-vertical">
                  <div class="el-scrollbar__thumb" style="height: 0px; transform: translateY(0%); -webkit-transform: translateY(0%);"></div>
                </div>
              </div>
              <!---->
            </div>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`Switch\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="switch" class="el-form-item__label">switch</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div role="switch" class="el-switch control"><input type="checkbox" name="switch" true-value="true" class="el-switch__input">
            <!----><span class="el-switch__core" style="width: 40px;"></span>
            <!---->
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`Textarea\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="textarea" class="el-form-item__label">textarea</label>
    <div class="el-form-item__content">
      <div class="former-top">
        <div>top</div>
      </div>
      <div class="former-wrapper">
        <div class="former-prefix">
          <div>prefix</div>
        </div>
        <div class="former-main">
          <div class="el-textarea control"><textarea autocomplete="off" aria-label="textarea" placeholder="textarea placeholder" class="el-textarea__inner" style="min-height: -4px;"></textarea></div>
        </div>
        <div class="former-suffix">
          <div>suffix</div>
        </div>
      </div>
      <div class="former-bottom">
        <div>bottom</div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`TimePicker\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="timePicker" class="el-form-item__label">timePicker</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--time control">
            <!----><input type="text" autocomplete="off" name="timePicker" class="el-input__inner"><span class="el-input__prefix"><i class="el-input__icon el-icon-time"></i><!----></span><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-input__icon"></i><!----></span>
            <!----></span>
            <!---->
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`TimeSelect\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="timeSelect" class="el-form-item__label">timeSelect</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--time-select control">
            <!----><input type="text" autocomplete="off" name="timeSelect" class="el-input__inner"><span class="el-input__prefix"><i class="el-input__icon el-icon-time"></i><!----></span><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-input__icon"></i><!----></span>
            <!----></span>
            <!---->
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`Tree\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="tree" class="el-form-item__label">tree</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div role="tree" class="el-tree control">
            <div role="treeitem" tabindex="0" draggable="false" class="el-tree-node is-focusable">
              <div class="el-tree-node__content" style="padding-left: 0px;"><span class="el-tree-node__expand-icon el-icon-caret-right"></span>
                <!---->
                <!----><span class="el-tree-node__label">foo</span></div>
              <!---->
            </div>
            <!---->
            <div class="el-tree__drop-indicator" style="display: none;"></div>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Former/form Render \`Upload\` component 1`] = `
<form class="el-form">
  <div class="el-form-item"><label for="upload" class="el-form-item__label">upload</label>
    <div class="el-form-item__content">
      <div class="former-wrapper">
        <div class="former-main">
          <div class="control">
            <div tabindex="0" class="el-upload el-upload--text">
              <div>
                <p>foo</p>
              </div><input type="file" name="upload" class="el-upload__input">
            </div><span tag="ul" name="el-list" class="el-upload-list el-upload-list--text"></span>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;
